src/main/java/com/synopsys/integration/alert/provider/blackduck/collector/builder/policy/PolicyClearedMessageBuilder.java

```diff
@@ -0,0 +1,156 @@
+            List<ComponentItem> items = new LinkedList<>();
+            for (Map.Entry<ComponentVersionStatus, Set<PolicyInfo>> componentToPolicyEntry : componentPolicies.entrySet()) {
+                ComponentVersionStatus componentVersionStatus = componentToPolicyEntry.getKey();
+                Set<PolicyInfo> policies = componentToPolicyEntry.getValue();
+                final List<ComponentItem> componentItems = retrievePolicyItems(responseCache, componentVersionStatus, policies, notificationId, operation, violationContent.getProjectVersion());
+                items.addAll(componentItems);
+            }
+            projectVersionMessageBuilder.applyAllComponentItems(items);
+            return List.of(projectVersionMessageBuilder.build());
+        } catch (AlertException ex) {
+            logger.error("Error creating policy violation cleared message.", ex);
+        }
+
+        return List.of();
+    }
+
+    private List<ComponentItem> retrievePolicyItems(BlackDuckResponseCache blackDuckResponseCache, ComponentVersionStatus componentVersionStatus,
+        Set<PolicyInfo> policies, Long notificationId, ItemOperation operation, String projectVersionUrl) {
+        List<ComponentItem> componentItems = new LinkedList<>();
+        String componentName = componentVersionStatus.getComponentName();
+        String componentVersionName = componentVersionStatus.getComponentVersionName();
+        ComponentData componentData = new ComponentData(componentName, componentVersionName, projectVersionUrl);
+        componentItems.addAll(policyCommonBuilder.retrievePolicyItems(blackDuckResponseCache, componentData, policies, notificationId, operation,
+            componentVersionStatus.getBomComponent(), List.of()));
+
+        for (PolicyInfo policyInfo : policies) {
+            LinkableItem policyNameItem = componentBuilderUtil.createPolicyNameItem(policyInfo);
+            Optional<PolicyRuleView> optionalPolicyRule = blackDuckResponseCache.getPolicyRule(blackDuckResponseCache, policyInfo);
+            if (optionalPolicyRule.isPresent() && policyCommonBuilder.hasVulnerabilityRule(optionalPolicyRule.get())) {
+                Optional<ComponentItem> vulnerabilityComponent = createEmptyVulnerabilityItem(blackDuckResponseCache, policyNameItem, componentData, notificationId, operation);
+                vulnerabilityComponent.ifPresent(componentItems::add);
+            }
+        }
+        return componentItems;
+    }
+
+    private Optional<ComponentItem> createEmptyVulnerabilityItem(BlackDuckResponseCache blackDuckResponseCache, LinkableItem policyNameItem, ComponentData componentData, Long notificationId,
+        ItemOperation operation) {
+        LinkableItem vulnerabilityItem = new LinkableItem(MessageBuilderConstants.LABEL_VULNERABILITIES, "ALL", null);
```

Comment:
I don't think you need to pass null here as the two param ctor will add null for you.